components/layout/display_list_builder.rs

```diff
@@ -1005,9 +1134,7 @@ impl FragmentDisplayListBuilding for Fragment {
         let mut clip = clip.clone();
         clip.intersect_rect(absolute_bounds);
 
-        let grad = self.convert_gradient(absolute_bounds, gradient, style);
-
-        if let Some(x) = grad {
+        if let Some(x) = self.convert_gradient(absolute_bounds, gradient, style) {
```

Comment:
This code can be simplified if `convert_gradient` and `convert_radial_gradient` are changed to accept the stops and kind values instead of complete gradients.